zerver/views/registration.py

```diff
@@ -223,17 +227,16 @@ def accounts_register(*args: Any, **kwargs: Any) -> HttpResponse:
     return registration_helper(*args, **kwargs)
 
 
-@has_request_variables
+@typed_endpoint
 def registration_helper(
     request: HttpRequest,
-    key: str = REQ(default=""),
-    timezone: str = REQ(default="", converter=to_timezone_or_empty),
-    from_confirmation: str | None = REQ(default=None),
-    form_full_name: str | None = REQ("full_name", default=None),
-    source_realm_id: int | None = REQ(
-        default=None, converter=to_converted_or_fallback(to_non_negative_int, None)
-    ),
-    form_is_demo_organization: str | None = REQ("is_demo_organization", default=None),
+    *,
+    key: str = "",
+    timezone: Annotated[str, timezone_or_empty_validator()] = "",
+    from_confirmation: str | None = None,
+    form_full_name: Annotated[str | None, ApiParamConfig("full_name")] = None,
+    source_realm_id: Annotated[int | None, non_negative_int_or_none_validator()] = None,
```

Comment:
Can you explain how the new validator behaves the same as the old one? I don't totally understand why `to_converted_or_fallback` existed....